controllers/volumereplicationgroup_controller_test.go

```diff
@@ -37,6 +38,102 @@ func init() {
 var UploadedPVs = make(map[string]interface{})
 
 var _ = Describe("Test VolumeReplicationGroup", func() {
+	conditionExpect := func(conditions []metav1.Condition, typ string) *metav1.Condition {
+		condition := meta.FindStatusCondition(conditions, typ)
+		Expect(condition).ToNot(BeNil())
+
+		return condition
+	}
+	conditionStatusReasonExpect := func(condition *metav1.Condition, status metav1.ConditionStatus, reason string) {
+		Expect(condition.Status).To(Equal(status))
+		Expect(condition.Reason).To(Equal(reason))
+	}
+	var vrg *ramendrv1alpha1.VolumeReplicationGroup
+	vrgConditionExpect := func(typ string) *metav1.Condition {
+		return conditionExpect(vrg.Status.Conditions, typ)
+	}
+	vrgConditionStatusReasonExpect := func(typ string, status metav1.ConditionStatus, reason string) *metav1.Condition {
+		condition := vrgConditionExpect(typ)
+		conditionStatusReasonExpect(condition, status, reason)
+
+		return condition
+	}
+	vrgGet := func() {
+		Expect(apiReader.Get(context.TODO(), types.NamespacedName{
+			Namespace: vrg.Namespace, Name: vrg.Name,
+		}, vrg)).To(Succeed())
+	}
+	var dataReadyCondition *metav1.Condition
+	When("ReplicationState is invalid", func() {
+		It("should set DataReady status=False reason=Error", func() {
```

Comment:
Checked the code and this seems to be a valid check (i.e. to expect condition.Status for DataReady condition to be false). Because, if the ReplicationState is invalid, then VRG will recognize that and will set the condition.Status for DataReady to false.